File: src/MyServiceBus.Abstractions/BusStopTimeoutException.cs
namespace MyServiceBus;

/// <summary>
/// Thrown when a bus cannot drain its active receive work within the configured shutdown timeout.
/// </summary>
public sealed class BusStopTimeoutException : TimeoutException
{
    public BusStopTimeoutException(TimeSpan timeout)
        : base($"The service bus did not stop within the configured timeout of {timeout}.")
    {
        Timeout = timeout;
    }

    public BusStopTimeoutException(TimeSpan timeout, Exception innerException)
        : base($"The service bus did not stop within the configured timeout of {timeout}.", innerException)
    {
        Timeout = timeout;
    }

    public TimeSpan Timeout { get; }
}

File: src/Java/myservicebus-abstractions/src/main/java/com/myservicebus/BusStopTimeoutException.java
package com.myservicebus;

import java.time.Duration;

/**
 * Indicates that active receive work did not drain within the configured bus stop timeout.
 */
public final class BusStopTimeoutException extends RuntimeException {
    private final Duration timeout;

    public BusStopTimeoutException(Duration timeout) {
        super("The service bus did not stop within the configured timeout of " + timeout);
        this.timeout = timeout;
    }

    public BusStopTimeoutException(Duration timeout, Throwable cause) {
        super("The service bus did not stop within the configured timeout of " + timeout, cause);
        this.timeout = timeout;
    }

    public Duration getTimeout() {
        return timeout;
    }
}
